PR: https://github.com/odoo/odoo/pull/258679

From: 00545bdd8774764ef53c3cef8f19a17e795bf923
From: Corentin Deruyck
Date: 2026-05-07 13:44:21

Breaking data model changes scores: del:3 + add:0, change matches:
-    _inherit = "res.company"
-    sale_order_template_id = fields.Many2one(
-    company_so_template_id = fields.Many2one(

Total Changes: 71

[IMP] sale,sale_management: Adding quotation template via new button

The goal is to add the ability to choose which template to use
when clicking the “New” button. The idea is to replicate what
has already been implemented in the Project application. When
clicking the “New” button and SO templates are available, users
will be able to select a template from that first click. It will
also be possible, directly from this menu, to navigate to the view
for editing the template. Additionally, users will be able to
delete the template from this menu.

Furthermore, the “Quotation Template” field has been moved to the
“Other Info” page, and three fields from the “Other Info” page
have been moved into a new column.

Finally, the Default Template option in the Sales app’s settings
has been removed because it duplicates the purpose of the task.

task-6088445

closes odoo/odoo#258679

Related: odoo/enterprise#113771
Related: odoo/upgrade#9913
Signed-off-by: Victor Feyens (vfe) <vfe@odoo.com>

================================= pseudo patch: =================================

--- a/addons/sale_management/models/__init__.py
+++ b/addons/sale_management/models/__init__.py
@@ -2,7 +2,6 @@
 
 from . import (
     digest,
-    res_company,
     res_config_settings,
     sale_order,
     sale_order_line,

--- a/addons/sale_management/models/res_company.py
+++ b/None
@@ -1,20 +0,0 @@
-# Part of Odoo. See LICENSE file for full copyright and licensing details.
-
-from odoo import fields, models
-
-
-class ResCompany(models.Model):
-    _inherit = "res.company"
-    _check_company_auto = True
-
-    sale_order_template_id = fields.Many2one(
-        string="Default Sale Template",
-        comodel_name="sale.order.template",
-        domain="""[
-            ('template_type', '=', 'quotation'),
-            '|',
-            ('company_id', '=', False),
-            ('company_id', '=', id),
-        ]""",
-        check_company=True,
-    )

--- a/addons/sale_management/models/res_config_settings.py
+++ b/addons/sale_management/models/res_config_settings.py
@@ -9,25 +9,3 @@ class ResConfigSettings(models.TransientModel):
     group_sale_order_template = fields.Boolean(
         string="Quotation Templates", implied_group="sale_management.group_sale_order_template"
     )
-    company_so_template_id = fields.Many2one(
-        string="Default Template",
-        related="company_id.sale_order_template_id",
-        readonly=False,
-        domain="""[
-            ('template_type', '=', 'quotation'),
-            '|',
-            ('company_id', '=', False),
-            ('company_id', '=', company_id),
-        ]""",
-    )
-
-    def set_values(self):
-        if not self.group_sale_order_template:
-            if self.company_so_template_id:
-                self.company_so_template_id = False
-            companies = (
-                self.env["res.company"].sudo().search([("sale_order_template_id", "!=", False)])
-            )
-            if companies:
-                companies.sale_order_template_id = False
-        super().set_values()

--- a/addons/sale_management/models/sale_order.py
+++ b/addons/sale_management/models/sale_order.py
@@ -12,12 +12,10 @@ class SaleOrder(models.Model):
 
     sale_order_template_id = fields.Many2one(
         comodel_name="sale.order.template",
-        string="Quotation Template",
-        compute="_compute_sale_order_template_id",
+        string="Template",
         store=True,
         readonly=False,
         check_company=True,
-        precompute=True,
         index='btree_not_null',
         domain="""[
             ('template_type', '=', 'quotation'),
@@ -29,17 +27,6 @@ class SaleOrder(models.Model):
 
     # === COMPUTE METHODS ===#
 
-    # Do not make it depend on `company_id` field
-    # It is triggered manually by the _onchange_company_id below iff the SO has not been saved.
-    def _compute_sale_order_template_id(self):
-        for order in self:
-            company_template = order.company_id.sale_order_template_id
-            if company_template and order.sale_order_template_id != company_template:
-                if "website_id" in self._fields and order.website_id:
-                    # don't apply quotation template for order created via eCommerce
-                    continue
-                order.sale_order_template_id = order.company_id.sale_order_template_id.id
-
     @api.depends("partner_id", "sale_order_template_id")
     def _compute_note(self):
         super()._compute_note()
@@ -82,14 +69,6 @@ class SaleOrder(models.Model):
 
     # === ONCHANGE METHODS ===#
 
-    @api.onchange("company_id")
-    def _onchange_company_id(self):
-        """Trigger quotation template recomputation on unsaved records company change."""
-        super()._onchange_company_id()
-        if self._origin.id:
-            return
-        self._compute_sale_order_template_id()
-
     @api.onchange("sale_order_template_id")
     def _onchange_sale_order_template_id(self):
         if not self.sale_order_template_id:

--- a/addons/sale_management/models/sale_order_template.py
+++ b/addons/sale_management/models/sale_order_template.py
@@ -209,11 +209,6 @@ class SaleOrderTemplate(models.Model):
         return records
 
     def write(self, vals):
-        if "active" in vals and not vals.get("active"):
-            companies = (
-                self.env["res.company"].sudo().search([("sale_order_template_id", "in", self.ids)])
-            )
-            companies.sale_order_template_id = None
         result = super().write(vals)
         self._update_product_translations()
         return result
